### DISCUSSION TITLE
Integrate Integrate Intelligent Octopus Go dynamic charging data for Powerwall Charging automations

### DISCUSSION THREAD

**Discussion by duffjp** [netzero-labs/netzero#37] [Ideas]
Hi,

Intelligent Octopus Go (IOG - https://octopus.energy/smart/intelligent-octopus-go/) is a smart tariff from Octopus that gives an off peak rate for car charging coupled with a peak rate for daytime usage.  There is a standard window of 6hrs over night, which the PW can make use of to charge using time based controls.  However, IOG also offers dynamic off peak slots during what would ordinarily be a peak time, which PW on its own can’t cope with.

The information about dynamic allocation of an off peak slot is available via an Octopus API.  I currently use it in Home Assistant, via this integration for Octopus:

https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy

It presents an entity in Home Assistant as Intelligent Dispatching. 

There is also another integration (https://github.com/megakid/ha_octopus_intelligent) that can present other details such as the time the current off-peak period is scheduled to end.

If you were able to integrate some of this info into NetZero, this would be hugely beneficial for many of your apps users.

Thanks,

Jon.

**Comment by zigam** [comment]
Thanks for the suggestion!

Netzero is currently using the public Octopus API, and the Intelligent Go extra periods are unfortunately not available in the public API. We may consider using user-provided API keys in the future.

There's a workaround some users use: if you create an automation to trigger when the EV is charging (e.g. with the Tesla Wall Connector integration or by using a "house usage > 7kW" rule), you can bump up the Powerwall backup reserve to 100% while the EV is charging. This will avoid discharging the Powerwall into the EV, and additionally charge it from the grid while electricity is cheap.

When the EV stops charging, you reset the backup reserve to your desired value.
